### PR TITLE
Test search permission #373

### PR DESCRIPTION
**Pull request by peteranny**
In the beginning, the permission is locked
Test for search permission after user signed in
If failed, update share link
If success, update `user_enabled` (which determines the style of search bar and search result)

`testSearchPermission()` is called by `loginStatusChange()` in `src/js/global/user.js`

### CODE REVIEW

File: src/js/global/user.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ const loginStatusChange = (response) => {
     FB.api('/me', (childResponse) => {
       changeLoginBlock(true, childResponse.name);
     });
+    if(testSearchPermission) testSearchPermission();
```
Comment (barry800414) [inline src/js/global/user.js]: 用 if(typeof testSearchPermission !== "undefined") ，當testSearchPermission真的沒定義的時候，才不會有ReferenceError  (可以連到index.html試試看) 

ref: http://stackoverflow.com/questions/1042138/javascript-check-if-function-exists

**Review by barry800414 — COMMENTED** [review]
其他部份看起來沒有問題。

File: src/js/main/show.js [hunk 2]
```diff
@@ -462,6 +462,32 @@ $(function(){
...
+    const hasSearchPermission = response.hasSearchPermission;
+    if(hasSearchPermission) changeUserEnabled(true);
+    else{
+      return $.ajax({
+        url: WTS.constants.backendURL + "me/recommendations",
+        method:'POST',
+        data: {access_token},
+        dataType: "json",
+      }).then(response => {
+        const recomm_url = `https://worktime.goodjob.life?rec_usr=${response.recommendation_string}`
```
Comment (mark86092) [inline src/js/main/show.js]: 我記得這裡有提到網址是：

https://github.com/goodjoblife/WorkTimeSurvey-backend/issues/143

image

 usr 不太好理解，還易誤會

BTW, 網址請用 WTS.constants 當中的變數替代

**Comment by peteranny** [comment]
Done

**Comment by barry800414** [comment]
噢...我就是找不到那份doc，才跟施亭屹說?rec_user

**Pull request merged**